src/Sarif.Viewer.VisualStudio/Controls/InfoBar.cs

```diff
@@ -0,0 +1,184 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Imaging.Interop;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Sarif.Viewer.Controls
+{
+    /// <summary>
+    /// Displays a banner with a message for the user.
+    /// </summary>
+    public class InfoBar : IVsInfoBarUIEvents
+    {
+        /// <summary>
+        /// Standard spacing between info bar text elements.
+        /// </summary>
+        public const string InfoBarTextSpacing = "   ";
+
+        private readonly object showLock = new object();
+
+        private readonly IVsInfoBarTextSpan[] content;
+        private readonly Action<IVsInfoBarActionItem> clickAction;
+        private readonly Action closeAction;
+        private readonly ImageMoniker imageMoniker;
+
+        private InfoBarModel infoBarModel;
+
+        private IVsInfoBarUIElement uiElement;
+        private uint eventCookie;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="InfoBar"/> class.
+        /// </summary>
+        /// <param name="text">
+        /// The text to dislay.
+        /// </param>
+        /// <param name="clickAction">
+        /// An action to take when a user clicks on an <see cref="IVsInfoBarActionItem"/> (e.g. button) in the info bar.
+        /// </param>
+        /// <param name="closeAction">
+        /// An action to take when the info bar is closed.
+        /// </param>
+        public InfoBar(string text, Action<IVsInfoBarActionItem> clickAction = null, Action closeAction = null, ImageMoniker imageMoniker = default(ImageMoniker))
```

Comment:
>imageMoniker [](start = 125, length = 12)

Gabe: And I added this ctor parameter. In the code I borrowed, the image was always the blue "information" circle. For our purposes, IMO "error" is a better default.